src/tito/builder/submodule_aware_builder.py

```diff
@@ -0,0 +1,210 @@
+# Copyright (c) 2008-2011 Red Hat, Inc.
+#
+# This software is licensed to you under the GNU General Public License,
+# version 2 (GPLv2). There is NO WARRANTY for this software, express or
+# implied, including the implied warranties of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+# along with this software; if not, see
+# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+#
+# Red Hat trademarks are not licensed under GPLv2. No permission is
+# granted to use or replicate Red Hat trademarks that are incorporated
+# in this software or its documentation.
+"""
+Tito builders for a variety of common methods of building sources, srpms,
+and rpms.
+"""
+
+import os
+
+from tito.builder import Builder
+from tito.common import debug, run_command, create_tgz, find_spec_like_file, get_commit_timestamp
+from tito.tar import TarFixer
+
+
+class PushDir(object):
+    """
+    Helper class to change directories and restore the same way pushd/popd work in the shell.
+    Usage:
+    # cwd = 'current'
+    with PushDir("into"):
+        # cwd = 'current/into'
+        do_command # cwd is into.
+    # cwd = 'current'
+    """
+
+    def __init__(self, subdir):
+        self.old_wd = os.getcwd()
+        self.new_wd = None
+        if subdir:
+            self.new_wd = os.path.join(self.old_wd, subdir)
+
+    def __enter__(self):
+        if self.new_wd:
+            self.chdir(self.old_wd, self.new_wd)
+
+    def __exit__(self, type, value, traceback):
+        if self.new_wd:
+            self.chdir(self.new_wd, self.old_wd)
+
+    def chdir(self, old_dir, new_dir):
+        if new_dir:
+            debug('Changing directory from %s to %s' % (old_dir, new_dir))
+            os.chdir(new_dir)
+
+
+class SubmoduleAwareBuilder(Builder):
+    """
+    Submodule aware builder class.
+
+    This builder is aware of submodules and will package the contents of the submodules as well.
+    It expectes `git submodule--helper list` to exist and work.
+    """
+    REQUIRED_ARGS = []
+
+    # TODO: drop version
+    def __init__(self, name=None, tag=None, build_dir=None,
```

Comment:
If I understand it correctly, this constructor just calls the parent's constructor and does nothing else?
In that case, we can remove it altogether :-)